azimuth_caas_operator/utils/cluster_type.py

```diff
@@ -45,12 +46,22 @@ async def _cache_client_type(client, cluster, cluster_type_spec, cluster_version
     cluster_resource = await client.api(registry.API_VERSION).resource(
         "clusters/status"
     )
+    patchedTimestamp = None
+    # if we have an existing version, add a patched timestamp
+    if cluster.status.clusterTypeVersion:
+        now = datetime.datetime.utcnow()
+        patchedTimestamp = now.strftime("%Y-%m-%dT%H:%M:%SZ")
```

Comment:
Not sure I understand this logic. Shouldn't we only set the `patchedTimestamp` when the `clusterTypeVersion` has changed?

Reply:
So this code is only ever called when the version as changed, this is just to skip adding the date when we first create the cluster, so we can tell if its ever been patched or not.